server/src/test/java/com/linecorp/centraldogma/server/internal/api/MergeFileTest.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.server.internal.api;
+
+import static net.javacrumbs.jsonunit.fluent.JsonFluentAssert.assertThatJson;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+
+public class MergeFileTest extends ContentServiceV1TestBase {
+
+    @Test
+    public void mergeJsonFiles() {
+        addFilesForMergeJson();
+
+        // The property "a" in "/foo.json" is overwritten by the property "a" in "/foo2.json"
+        String queryString = "path=/foo.json" + '&' +
+                             "path=/foo1.json" + '&' +
+                             "path=/foo2.json" + '&' +
+                             "optional_path=/foo3.json";
+
+        AggregatedHttpMessage aRes = httpClient().get("/api/v1/projects/myPro/repos/myRepo/merge?" +
+                                                      queryString).aggregate().join();
+
+        final String expectedJson =
+                '{' +
+                "   \"revision\" : 4," +
+                "   \"type\" : \"JSON\"," +
+                "   \"content\" : {" +
+                "       \"a\" : \"new_bar\"," +
+                "       \"b\" : \"baz\" " +
+                "   }," +
+                "   \"paths\" : [\"/foo.json\", \"/foo1.json\", \"/foo2.json\"] " +
+                '}';
+        final String actualJson = aRes.content().toStringUtf8();
+        assertThatJson(actualJson).isEqualTo(expectedJson);
+
+        queryString = "path=/foo.json" + '&' +
+                      "path=/foo1.json" + '&' +
+                      "path=/foo2.json" + '&' +
+                      "path=/foo3.json";
+        aRes = httpClient().get("/api/v1/projects/myPro/repos/myRepo/merge?" + queryString).aggregate()
+                           .join();
+        assertThat(aRes.status()).isEqualTo(HttpStatus.NOT_FOUND);
+        assertThat(aRes.content().toStringUtf8()).contains("Entry '/foo3.json (4)' does not exist.");
```

Comment:
Global comment: Could you make sure the failure is handled by `HttpApiExceptionHandler` so that the error response is also JSON?